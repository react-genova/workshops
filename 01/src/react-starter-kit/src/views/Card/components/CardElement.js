//@flow
import * as React from 'react';
import styled from 'styled-components';
import LabelLink from '../../LabelLink'

type Props = {
    title:string,
    description:string,
    url:string,
    onClick:(url:string)=>void,
    children?:React.Node,
};

const Container = styled.div``;

const Indenter = styled.div`
    padding-left: 1.3em;
`;

const CardElement = ({title, description, url, onClick, children}:Props) => (
    <Container>
        <LabelLink title={title} description={description} url={url} onClick={onClick} />
        <Indenter>{children}</Indenter>
    </Container>
);

export default CardElement;