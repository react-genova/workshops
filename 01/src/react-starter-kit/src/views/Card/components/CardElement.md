### Simple CardElement ###

```js
<CardElement title="Google" description="One of the best search engine" onClick={()=>null} url="https://www.google.it/" />
```

### Nested CardElements ###

```js
const onClick = ()=>null;
<CardElement title="Search engines" description="A not  exhaustive list of search engines" url="" onClick={onClick}>
    <CardElement title="Google" description="One of the best search engine" url="https://www.google.it/" onClick={onClick}/>
    <CardElement title="Duck Duck Go" description="Another search engine" url="https://duckduckgo.com/" onClick={onClick}/>
</CardElement>
```

### Very Nested CardElements ###

```js
const onClick = ()=>null;
<CardElement title="Most popular sites" description="" url="" onClick={onClick}>
    <CardElement title="e-commerce" description="" url="" onClick={onClick}>
        <CardElement title="Amazon" description="" url="https://www.amazon.it/" onClick={onClick}/>
        <CardElement title="Alibaba" description="" url="https://www.alibaba.com/" onClick={onClick}/>
    </CardElement>
    <CardElement title="Search engines" description="" url="" onClick={onClick}>
        <CardElement title="Google" description="" url="https://www.google.it/" onClick={onClick}/>
        <CardElement title="Duck Duck Go" description="" url="https://duckduckgo.com/" onClick={onClick}/>
    </CardElement>
</CardElement>
```