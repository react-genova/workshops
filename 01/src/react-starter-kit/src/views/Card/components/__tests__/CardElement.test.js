//@flow
import React from 'react';
import CardElement from '../CardElement';
import { shallow } from 'enzyme';

describe('CardElement component', () => {

    it('simple CardElement snapshot', () => {
        const onClick = (url:string)=>{url.toLocaleLowerCase();};
        const component = shallow(<CardElement title="Google" description="One of the best search engine" onClick={onClick} url="https://www.google.it/" />);
        expect(component).toMatchSnapshot();
    });

    it('nested CardElement snapshot', () => {
        const onClick = (url:string)=>{url.toLocaleLowerCase();};
        const component = shallow(
            <CardElement title="Search engines" description="A not  exhaustive list of search engines" url="" onClick={onClick}>
                <CardElement title="Google" description="One of the best search engine" url="https://www.google.it/" onClick={onClick}/>
                <CardElement title="Duck Duck Go" description="Another search engine" url="https://duckduckgo.com/" onClick={onClick}/>
            </CardElement>
        );
        expect(component).toMatchSnapshot();
    });

});