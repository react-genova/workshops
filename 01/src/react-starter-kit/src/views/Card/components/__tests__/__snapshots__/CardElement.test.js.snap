// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardElement component nested CardElement snapshot 1`] = `
<styled.div>
  <LabelLink
    description="A not  exhaustive list of search engines"
    onClick={[Function]}
    title="Search engines"
    url=""
  />
  <styled.div>
    <CardElement
      description="One of the best search engine"
      onClick={[Function]}
      title="Google"
      url="https://www.google.it/"
    />
    <CardElement
      description="Another search engine"
      onClick={[Function]}
      title="Duck Duck Go"
      url="https://duckduckgo.com/"
    />
  </styled.div>
</styled.div>
`;

exports[`CardElement component simple CardElement snapshot 1`] = `
<styled.div>
  <LabelLink
    description="One of the best search engine"
    onClick={[Function]}
    title="Google"
    url="https://www.google.it/"
  />
  <styled.div />
</styled.div>
`;
